x/minfee/grpc_query.go

```diff
@@ -0,0 +1,34 @@
+package minfee
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params/keeper"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var _ QueryServer = &QueryServerImpl{}
+
+// QueryServerImpl wraps the params keeper and implements the minfee gRPC query server.
+type QueryServerImpl struct {
+	paramsKeeper keeper.Keeper
+}
+
+// NewQueryServerImpl creates a new QueryServerImpl.
+func NewQueryServerImpl(paramsKeeper keeper.Keeper) *QueryServerImpl {
+	return &QueryServerImpl{paramsKeeper: paramsKeeper}
+}
+
+// NetworkMinGasPrice returns the global minimum gas price.
+func (q *QueryServerImpl) NetworkMinGasPrice(ctx context.Context, _ *QueryNetworkMinGasPrice) (*QueryNetworkMinGasPriceResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	var params Params
+	subspace, found := q.paramsKeeper.GetSubspace(ModuleName)
+	if !found {
+		return nil, status.Errorf(codes.NotFound, "subspace not found for minfee. Minfee is only active v2 and onwards")
+	}
+	subspace.GetParamSet(sdkCtx, &params)
+	return &QueryNetworkMinGasPriceResponse{NetworkMinGasPrice: params.GlobalMinGasPrice}, nil
```

Comment:
nit: I personally slightly prefer NetworkMinGasPrice but I think it's more important if all the terms related to this param were consistent. 

In other words, `params.GlobalMinGasPrice` should also be renamed to `params.NetworkMinGasPrice`. If we're not doing that, then this PR should revert back to the previous name GlobalMinGasPrice so that the codebase is consistent and uses one term for it.

Reply:
Yes I agree. I was planning to do that in a follow up PR

Reply:
I also like NetworkMinGasPrice better